packages/fleather/lib/src/services/clipboard_manager.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/services.dart';
+import 'package:parchment_delta/parchment_delta.dart';
+
+/// Encapsulates clipboard data
+///
+/// One of the properties should be non-null or null should be returned
+/// from [FleatherCustomClipboardGetData].
+/// When pasting data in editor, [delta] has precedence over [plainText].
+class FleatherClipboardData {
+  final String? plainText;
+  final Delta? delta;
```

Comment:
Don't we want to had possibility to paste HTML?

Reply:
IMHO, we should never add anything other than delta and plainText. Having html will add the complexity of handling custom embeds (e.g. image, video, ...). Also in case of implementing rich clipboard in Fleather, we need to handle other clipboard data like images. Last but not least, the differences in platforms and how they represent data in clipboard will introduce more complexity.

I think we can easily avoid all of this by providing users with 'FleatherCustomClipboardManager' and a decent html codec in Parchment. WDYT?